File: Pirate.Lexer.Test/KeyWordServiceTest.cs
using Pirate.Lexer.Enums;
using Pirate.Lexer.Tokens;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Xunit;

namespace Pirate.Lexer.Test
{
    public class KeyWordServiceTest
    {
        [Fact]
        public void ShouldReturnVar()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("var");

            Assert.Equal(TokenType.VAR, result);
        }

        [Fact]
        public void ShouldReturnInt()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("int");

            Assert.Equal(TokenType.INT, result);
        }

        [Fact]
        public void ShouldReturnFloat()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("float");

            Assert.Equal(TokenType.FLOAT, result);
        }

        [Fact]
        public void ShouldReturnString()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("string");

            Assert.Equal(TokenType.STRING, result);
        }

        [Fact]
        public void ShouldReturnChar()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("char");

            Assert.Equal(TokenType.CHAR, result);
        }

        [Fact]
        public void ShouldReturnIf()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("if");

            Assert.Equal(TokenType.IF, result);
        }

        [Fact]
        public void ShouldReturnElse()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("else");

            Assert.Equal(TokenType.ELSE, result);
        }

        [Fact]
        public void ShouldReturnFor()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("for");

            Assert.Equal(TokenType.FOR, result);
        }

        [Fact]
        public void ShouldReturnTo()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("to");

            Assert.Equal(TokenType.TO, result);
        }

        [Fact]
        public void ShouldReturnForeach()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("foreach");

            Assert.Equal(TokenType.FOREACH, result);
        }

        [Fact]
        public void ShouldReturnIn()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("in");

            Assert.Equal(TokenType.IN, result);
        }

        [Fact]
        public void ShouldReturnWhile()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("while");

            Assert.Equal(TokenType.WHILE, result);
        }

        [Fact]
        public void ShouldReturnFunc()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("func");

            Assert.Equal(TokenType.FUNC, result);
        }

        [Fact]
        public void ShouldReturnClass()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("class");

            Assert.Equal(TokenType.CLASS, result);
        }

        [Fact]
        public void ShouldReturnNew()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("new");

            Assert.Equal(TokenType.NEW, result);
        }

        [Fact]
        public void ShouldReturnVoid()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTypeKeyword("void");

            Assert.Equal(TokenType.VOID, result);
        }

        [Fact]
        public void ShouldReturnTrue()
        {
            var keyWordService = new KeyWordService();
            var result = keyWordService.GetTokenControlKeyword("return");

            Assert.Equal(TokenType.RETURN, result);
        }
    }
}

File: src/Pirate.Lexer.F.Test/KeyWordServiceTest.fs
module Pirate.Lexer.Test.KeyWordServiceTest

open Xunit
open Pirate.Lexer
open Pirate.Lexer.Enums


[<Fact>]
let ShouldReturnVar () =
    Assert.Equal(
        VAR, 
        KeyWordService().GetTypeKeyword("var")
    )

[<Fact>]
let ShouldReturnInt () = 
    Assert.Equal(
        INT, 
        KeyWordService().GetTypeKeyword("int")
    )

[<Fact>]
let ShouldReturnFloat () = 
    Assert.Equal(
        FLOAT, 
        KeyWordService().GetTypeKeyword("float")
    )

[<Fact>]
let ShouldReturnString () = 
    Assert.Equal(
        STRING, 
        KeyWordService().GetTypeKeyword("string")
    )

[<Fact>]
let ShouldReturnChar () = 
    Assert.Equal(
        CHAR, 
        KeyWordService().GetTypeKeyword("char")
    )

[<Fact>]
let ShouldReturnIf () = 
    Assert.Equal(
        IF, 
        KeyWordService().GetTokenControlKeyword("if")
    )

[<Fact>]
let ShouldReturnElse () = 
    Assert.Equal(
        ELSE, 
        KeyWordService().GetTokenControlKeyword("else")
    )

[<Fact>]
let ShouldReturnFor () = 
    Assert.Equal(
        FOR, 
        KeyWordService().GetTokenControlKeyword("for")
    )

[<Fact>]
let ShouldReturnTo () = 
    Assert.Equal(
        TO, 
        KeyWordService().GetTokenControlKeyword("to")
    )

[<Fact>]
let ShouldReturnForeach () = 
    Assert.Equal(
        FOREACH, 
        KeyWordService().GetTokenControlKeyword("foreach")
    )

[<Fact>]
let ShouldReturnIn () = 
    Assert.Equal(
        IN, 
        KeyWordService().GetTokenControlKeyword("in")
    )

[<Fact>]
let ShouldReturnWhile () = 
    Assert.Equal(
        WHILE, 
        KeyWordService().GetTokenControlKeyword("while")
    )

[<Fact>]
let ShouldReturnFunc () = 
    Assert.Equal(
        FUNC, 
        KeyWordService().GetTokenControlKeyword("func")
    )

[<Fact>]
let ShouldReturnClass () = 
    Assert.Equal(
        CLASS, 
        KeyWordService().GetTokenControlKeyword("class")
    )

[<Fact>]
let ShouldReturnNew () = 
    Assert.Equal(
        NEW, 
        KeyWordService().GetTokenControlKeyword("new")
    )

[<Fact>]
let ShouldReturnVoid () = 
    Assert.Equal(
        VOID, 
        KeyWordService().GetTypeKeyword("void")
    )

[<Fact>]
let ShouldReturnReturn () = 
    Assert.Equal(
        RETURN, 
        KeyWordService().GetTokenControlKeyword("return")
    )
    